### ISSUE TITLE
How to use range_includes/range_not_includes for terraform_compute_firewall

### ISSUE THREAD

**Issue by jbrule** [bridgecrewio/checkov#5657]
**Describe the issue**
I am working on a policy to block ports 21,23,25,110 on [google_compute_firewall resources](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_firewall). My issue is how to inspect a firewall rule with a [port range specified](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_firewall#ports). I see that a [class exists for inspecting](https://github.com/bridgecrewio/checkov/blob/f20abfc4d2730fa61cde501f2967aeed9d09a1a6/checkov/terraform/checks/resource/gcp/AbsGoogleComputeFirewallUnrestrictedIngress.py#L30) this with resource based checks. 

## Questions
- How can the range check be accomplished with Graph based checks? 
- I also had an issue where the directory scan did not catch a violation in a nested firewall resource. Is there something I am doing wrong there? The terraform_plan scan did catch it.

Thank you

**Examples**
```hcl
resource "google_compute_firewall" "sp113_1_and_2_violating_firewall_4" {
    name = "sp113-violating-firewall-4"
    network = google_compute_network.sp113_violating_network.name

    # Allow connections to Memcached instance on this network
    allow{
      protocol = "udp"
      ports = ["11211"]
    }

    allow {
      protocol = "tcp"
      ports = ["21", "22", "23", "110"]
    }
}

# See terraform_plan output at end of issue
resource "google_compute_firewall" "sp113_1_violating_port_range_firewall_5" {
    name = "sp113-violating-firewall-5"
    network = google_compute_network.sp113_violating_network.name
    allow {
      protocol = "tcp"
      ports = ["443"]
    }
    allow {
      protocol = "tcp"
      ports = ["20-110"]
    }
}

module "sp113_1_violating_sensitive_port_firewall_module" {
  source  = "registryaddress/dir/m-network-firewall/google"
  version = "2.0.0"

  project_id           = "fake-project-id"
  network              = google_compute_network.sp113_violating_network.name
  action               = "allow"
  fw_source            = "iap"
  fw_destination       = "all"
  fw_purpose           = "testing"
  direction            = "INGRESS"
  description          = "Allow iap for api"
  ranges               = ["10.10.40.0/20"]
  use_service_accounts = false
  sources              = []
  rules = [
    {
      protocol = "tcp"
      ports    = [21,23]
    }
  ]
  extra_attributes = {
    enable_logging = true
    priority       = 1000
  }
}
# See module hcl below
```

My Policy
```yaml
definition:
  and:
    - not:
      - or:
        - cond_type: "attribute" #This works when the port is explicitly defined but will miss ranges
          resource_types:
          - "google_compute_firewall"
          attribute: "allow.*.ports"
          operator: "intersects"
          value: "23"
        - cond_type: "attribute" #This doesn't appear to be inspecting ranges
          resource_types:
          - "google_compute_firewall"
          attribute: "allow.*.ports"
          operator: "range_includes"
          value: "23"
```
Check Result. Expectation is these should all fail
```
Check: GCP_SP113_1: "The following ports are not allowed in firewall rules. FTP, Telnet, SMTP, POP3"
	PASSED for resource: google_compute_firewall.sp113_1_violating_port_range_firewall_5
	File: /sp113.tf:93-104
Check: GCP_SP113_1: "The following ports are not allowed in firewall rules within. FTP, Telnet, SMTP, POP3"
	FAILED for resource: google_compute_firewall.sp113_1_and_2_violating_firewall_4
	File: /sp113.tf:75-91
Check: GCP_SP113_1: "The following ports are not allowed in firewall rules. FTP, Telnet, SMTP, POP3"
	PASSED for resource: module.sp113_1_violating_sensitive_port_firewall_module.google_compute_firewall.custom[0]
	File: /.external_modules/registryaddress/dir/m-network-firewall/google/2.0.0/main.tf:21-64
        # ^ terraform_plan check does fail this 
```

**Additional context**
The Python based checks expand the ranges
https://github.com/bridgecrewio/checkov/blob/f20abfc4d2730fa61cde501f2967aeed9d09a1a6/checkov/terraform/checks/resource/gcp/AbsGoogleComputeFirewallUnrestrictedIngress.py#L30

## Module Resource
Module
```hcl
locals {
  firewall_name = lower(join("-", list(var.fw_source, var.fw_destination, var.fw_purpose, var.action)))
}

resource "google_compute_firewall" "custom" {
  count                   = var.enabled ? 1 : 0
  name                    = local.firewall_name
  description             = var.description
  direction               = var.direction
  network                 = var.network
  project                 = var.project_id
  source_ranges           = var.direction == "INGRESS" ? var.ranges : null
  destination_ranges      = var.direction == "EGRESS" ? var.ranges : null
  source_tags             = var.use_service_accounts || var.direction == "EGRESS" ? null : var.sources
  source_service_accounts = var.use_service_accounts && var.direction == "INGRESS" ? var.sources : null
  target_tags             = var.use_service_accounts ? null : var.targets
  target_service_accounts = var.use_service_accounts ? var.targets : null
  disabled                = lookup(var.extra_attributes, "disabled", false)
  priority                = lookup(var.extra_attributes, "priority", 1000)
 #'enable_logging' is now deprecated in favor of 'log_config' block 
 #enable_logging          = lookup(var.extra_attributes, "enable_logging", true)

  dynamic "log_config" {
    for_each = var.log_config != null ? [var.log_config] : []
    content {
      metadata = lookup(log_config.value, "metadata", null)
    }
  }

  dynamic "allow" {
    for_each = [for rule in var.rules : rule if var.action == "allow"]
    iterator = rule
    content {
      protocol = rule.value.protocol
      ports    = rule.value.ports
    }
  }

  dynamic "deny" {
    for_each = [for rule in var.rules : rule if var.action == "deny"]
    iterator = rule
    content {
      protocol = rule.value.protocol
      ports    = rule.value.ports
    }
  }

}
```
google_compute_firewall.sp113_1_violating_port_range_firewall_5 plan as appears in terraform_plan
```json
{
          "address": "google_compute_firewall.sp113_1_violating_port_range_firewall_5",
          "mode": "managed",
          "type": "google_compute_firewall",
          "name": "sp113_1_violating_port_range_firewall_5",
          "provider_name": "registry.terraform.io/hashicorp/google",
          "schema_version": 1,
          "values": {
            "allow": [
              {
                "ports": [
                  "20-110"
                ],
                "protocol": "tcp"
              },
              {
                "ports": [
                  "443"
                ],
                "protocol": "tcp"
              }
            ],
            "deny": [],
            "description": null,
            "disabled": null,
            "log_config": [],
            "name": "sp113-violating-firewall-5",
            "network": "sp113-violating-network",
            "priority": 1000,
            "source_service_accounts": null,
            "source_tags": null,
            "target_service_accounts": null,
            "target_tags": null,
            "timeouts": null
          }
        },
```

**Comment by jbrule** [comment]
I was able to get the following to work.

```yaml
definition:
  or:
    - cond_type: "attribute"
      resource_types:
      - "google_compute_firewall"
      attribute: "disabled"
      operator: "is_true"
    - and:
      - cond_type: "attribute"
        resource_types:
        - "google_compute_firewall"
        attribute: "allow[*].ports[*]"
        operator: "jsonpath_range_not_includes"
        value: "21"
      - cond_type: "attribute"
        resource_types:
        - "google_compute_firewall"
        attribute: "allow[*].ports[*]"
        operator: "jsonpath_range_not_includes"
        value: "23"
      - cond_type: "attribute"
        resource_types:
        - "google_compute_firewall"
        attribute: "allow[*].ports[*]"
        operator: "jsonpath_range_not_includes"
        value: "25"
      - cond_type: "attribute"
        resource_types:
        - "google_compute_firewall"
        attribute: "allow[*].ports[*]"
        operator: "jsonpath_range_not_includes"
        value: "110"
```
